websites/A/AniMixPlay/presence.ts

```diff
@@ -1,145 +1,123 @@
+  }
+} else if (new RegExp("^/v.").test(document.location.pathname)) {
+  if (iFrameVideo) {
+    [, endTimestamp] = presence.getTimestamps(
+      Math.floor(currentTime),
+      Math.floor(duration)
+    );
+  } else {
+    video = document.querySelector<HTMLVideoElement>(
+      "#playercontainer > div > div.plyr__video-wrapper > video"
+    );
+    ({ currentTime, duration, paused } = video);
+    [, endTimestamp] = presence.getTimestamps(
+      Math.floor(currentTime),
+      Math.floor(duration)
+    );
+  }
+    const title = document.querySelector(
         "#aligncenter > span.animetitle"
-      ).textContent;
+      ).textContent,
+      episode = document
+        .querySelector("#eptitle > span#eptitleplace")
+        .textContent.replace(/\D/g, "");
 
-      if (!isNaN(duration)) {
-        presenceData.smallImageKey = paused ? "pause-v1" : "play-v1";
-        presenceData.smallImageText = paused
-          ? (await strings).pause
-          : (await strings).play;
-        presenceData.startTimestamp = timestamps[0];
-        presenceData.endTimestamp = timestamps[1];
-        presenceData.details = "Currently watching...";
-        presenceData.state = title;
+  if (!isNaN(duration)) {
+    presenceData.smallImageKey = paused ? "pause-v1" : "play-v1";
+    presenceData.smallImageText = paused
+      ? (await strings).pause
+      : (await strings).play;
+    presenceData.endTimestamp = endTimestamp;
+    presenceData.details = "Currently watching...";
+    presenceData.state = `${title} - Ep. ${episode}`;
 
-        if (paused) {
-          delete presenceData.startTimestamp;
-          delete presenceData.endTimestamp;
-        }
-      }
-    } else if (document.location.pathname.includes("/anime/")) {
-      animepagetitle = document.querySelector("#animepagetitle").textContent;
-      animepagetype = document
-        .querySelector("#addInfo")
-        .textContent.split(" ")[5]
-        .trim(-1);
-      presenceData.details = "Currently reading...";
-      presenceData.state = animepagetitle + " (" + animepagetype + ")";
-      presenceData.smallImageKey = "reading-v1";
-      presenceData.smallImageText = "Reading...";
+    if (paused) {
+      delete presenceData.startTimestamp;
+      delete presenceData.endTimestamp;
     }
   }
+} else if (document.location.pathname.includes("/anime/")) {
+  const animepagetitle =
+      document.querySelector("#animepagetitle").textContent,
+    animepagetype = document
+      .querySelector("#addInfo")
+      .textContent.split(" ")[5]
+      .trim();
+  presenceData.details = "Currently reading...";
+  presenceData.state = `${animepagetitle} (${animepagetype})`;
+  presenceData.smallImageKey = "reading-v1";
+  presenceData.smallImageText = "Reading...";
+}
 
-  if (presenceData.details == null) {
-    presence.setTrayTitle();
-    presence.setActivity();
-  } else {
-    presence.setActivity(presenceData);
-  }
+if (presenceData.details === null) {
```

Comment:
```suggestion
if (!presenceData.details) {
```

Because it wouldn't be null?

Reply:
meh leave it like that, if something I will change it in #3834